datasync/chaindatafetcher/repository.go

```diff
@@ -30,3 +35,19 @@ type Repository interface {
 
 	SetComponent(component interface{})
 }
+
+type retry func(blockchain.ChainEvent) error
+
+func retryFunc(f retry) retry {
+	return func(event blockchain.ChainEvent) error {
+		var err error
+		for {
+			if err = f(event); err == nil {
+				break
+			}
+			logger.Warn("retrying...", "blockNumber", event.Block.NumberU64())
+			time.Sleep(DBInsertRetryInterval)
+		}
+		return err
```

Comment:
Return value is always `nil`.

Reply:
bff768b877137ef894c6be26221d1c4ec0621f34